### PR TITLE
Added SDL_CleanupEvent()

### PR DESCRIPTION
**Pull request by slouken**
This is used to free any dynamically allocated memory in events.

### CODE REVIEW

**Comment by icculus** [comment]
We have a bug open somewhere for TEXTINPUT event data that is too large for the event's buffer; can we do the same short_data magic for this event, too?

**Comment by slouken** [comment]
> We have a bug open somewhere for TEXTINPUT event data that is too large for the event's buffer; can we do the same short_data magic for this event, too?

That was fixed by chunking the event delivery for that event. That issue is closed.

File: include/SDL3/SDL_events.h [hunk 1]
```diff
@@ -257,44 +256,35 @@ typedef struct SDL_KeyboardEvent
     SDL_Keysym keysym;  /**< The key that was pressed or released */
 } SDL_KeyboardEvent;
 
-#define SDL_TEXTEDITINGEVENT_TEXT_SIZE (32)
+#define SDL_TEXTEDITINGEVENT_TEXT_SIZE 64
```
Comment (sezero) [inline include/SDL3/SDL_events.h:259]: This can cause trouble in sdl2-compat

File: include/SDL3/SDL_events.h [hunk 2]
```diff
@@ -257,44 +256,35 @@ typedef struct SDL_KeyboardEvent
...
-    Uint32 type;                                /**< ::SDL_EVENT_TEXT_EDITING_EXT */
-    Uint64 timestamp;                           /**< In nanoseconds, populated using SDL_GetTicksNS() */
-    SDL_WindowID windowID;                        /**< The window with keyboard focus, if any */
-    char* text;                                 /**< The editing text, which should be freed with SDL_free(), and will not be NULL */
-    Sint32 start;                               /**< The start cursor of selected editing text */
-    Sint32 length;                              /**< The length of selected editing text */
-} SDL_TextEditingExtEvent;
-
-#define SDL_TEXTINPUTEVENT_TEXT_SIZE (32)
+#define SDL_TEXTINPUTEVENT_TEXT_SIZE 64
```
Comment (sezero) [inline include/SDL3/SDL_events.h:276]: ... so can this.

**Pull request merged**